library/pkwrite.c

```diff
@@ -575,6 +575,9 @@ int mbedtls_pk_write_pubkey_pem( mbedtls_pk_context *key, unsigned char *buf, si
         return( ret );
     }
 
+    /* Clean any remaining der data from the buffer */
+    memset( buf + olen, 0, size - olen );
```

Comment:
I didn't twig earlier, but: why add after every call to `mbedtls_pem_write_buffer` instead of only once in `mbedtls_pem_write_buffer`? Doing it in `mbedtls_pem_write_buffer` makes the code more robust and smaller.